tests/runner/testresource.go

```diff
@@ -0,0 +1,101 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package runner
+
+import (
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Disposable is an interface representing the disposable test resources
+type Disposable interface {
+	Name() string
+	Init() error
+	Dispose() error
+}
+
+// TestResources holds initial resources and active resources
+type TestResources struct {
+	resources           []Disposable
+	resourcesLock       sync.Mutex
+	activeResources     []Disposable
+	activeResourcesLock sync.Mutex
+}
+
+// Add adds Disposable resource to resources queue
+func (r *TestResources) Add(dr Disposable) {
+	r.resourcesLock.Lock()
+	defer r.resourcesLock.Unlock()
+	r.resources = append(r.resources, dr)
+}
+
+// dequeueResource dequeus Disposable resource from resources queue
+func (r *TestResources) dequeueResource() Disposable {
```

Comment:
may be just "Remove" as method name..

Reply:
this method is to dequeue the resource from resources queue. not just remove the resource.